Controller/Annotations/AbstractScalarParam.php

```diff
@@ -49,10 +51,17 @@ public function getConstraints()
                 ),
             ));
         } elseif (is_array($this->requirements) && isset($this->requirements['rule']) && $this->requirements['error_message']) {
+            @trigger_error('Using the "rule" and "error_message" options as requirements is deprecated. Use an array of constraints instead.', E_USER_DEPRECATED);
             $constraints[] = new Regex(array(
                 'pattern' => '#^(?:'.$this->requirements['rule'].')$#xsu',
                 'message' => $this->requirements['error_message'],
             ));
+        } elseif (is_array($this->requirements)) {
+            foreach ($this->requirements as $requirement) {
+                if ($requirement instanceof Constraint) {
+                    $constraints[] = $requirement;
+                }
```

Comment:
```suggestion
                } else {
                    @trigger_error('Using an array not only containing `Constraint`s is deprecated since version 2.6.', E_USER_DEPRECATED);
                }
```